File: src/main/java/bot/Deployer.java
package bot;

import lombok.extern.log4j.Log4j2;
import org.telegram.telegrambots.ApiContextInitializer;
import org.telegram.telegrambots.meta.TelegramBotsApi;
import org.telegram.telegrambots.meta.exceptions.TelegramApiException;

import java.io.IOException;
import java.net.ServerSocket;
import java.net.Socket;

@Log4j2
public class Deployer {
    public static void main(String[] args) {
        log.info("App started");
        ApiContextInitializer.init();

        TelegramBotsApi botsApi = new TelegramBotsApi();

        try {
            botsApi.registerBot(new Rosmira());
        } catch (TelegramApiException e) {
            e.printStackTrace();
        }

        try (ServerSocket serverSocket = new ServerSocket(Integer.parseInt(System.getenv("PORT")))) {
            while (true) {
                //Hack for Heroku, need reworking
                Socket clientSocket = serverSocket.accept();
                clientSocket.getOutputStream().write(200);
                clientSocket.close();
                log.info("Socket closed: " + clientSocket.toString());
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
        log.info("App down");
    }
}

File: src/main/kotlin/com/rosmira/rosmiracdebot/bot/Deployer.kt
package com.rosmira.rosmiracdebot.bot

import org.apache.logging.log4j.kotlin.Logging
import org.springframework.context.event.ContextRefreshedEvent
import org.springframework.context.event.EventListener
import org.springframework.stereotype.Component
import org.telegram.telegrambots.ApiContextInitializer
import org.telegram.telegrambots.bots.DefaultBotOptions
import org.telegram.telegrambots.meta.TelegramBotsApi
import org.telegram.telegrambots.meta.exceptions.TelegramApiException

@Component
class Deployer : Logging {
    @EventListener
    fun startApplication(event: ContextRefreshedEvent) {
        try {
            logger.info("App started")
            ApiContextInitializer.init()

            val botsApi = TelegramBotsApi()

            try {
                botsApi.registerBot(RosmiraCdeBot(DefaultBotOptions()))
            } catch (e: TelegramApiException) {
                logger.error("Cannot register bot: ", e)
            }
        } catch (e: Exception) {
            logger.error("APP CRASHED: ", e)
        }
    }
}
